webapi/src/main/scala/org/knora/webapi/responders/v2/OntologyResponderV2.scala

```diff
@@ -803,17 +806,20 @@ class OntologyResponderV2 extends Responder {
     private def getNamedGraphEntityInfoV2ForNamedGraphV2(namedGraphIri: IRI, userProfile: UserProfileV1): Future[NamedGraphEntityInfoV2] = {
         for {
             cacheData <- getCacheData
-            response = NamedGraphEntityInfoV2(
-                namedGraphIri = namedGraphIri,
-                propertyIris = cacheData.namedGraphProperties(namedGraphIri),
-                resourceClasses = cacheData.namedGraphResourceClasses(namedGraphIri)
-            )
-        } yield response
+
+            _ = if (!cacheData.namedGraphs.contains(namedGraphIri)) {
+                throw InconsistentTriplestoreDataException(s"Named graph not found: $namedGraphIri")
+            }
+        } yield NamedGraphEntityInfoV2(
+            namedGraphIri = namedGraphIri,
+            propertyIris = cacheData.namedGraphProperties.getOrElse(namedGraphIri, Set.empty[IRI]),
+            resourceClasses = cacheData.namedGraphResourceClasses.getOrElse(namedGraphIri, Set.empty[IRI])
```

Comment:
Could there be such as a thing as a named graph not containing any resource class definitions?

Reply:
I'm not sure. Definitely there could be an ontology with no property definitions, because they're all inherited from another ontology.

Reply:
Ok, so it makes sense to return an empty Set in case it is empty.